Frontend build tooling spike: add Vite

Add Vite as the frontend build tool and validate that the existing app can be served through it. This is Step 3 of the expansion plan discussed in #18.

**Scope:** Build tooling only — no behavior change, no framework yet.

**Proposed approach:**
- Install Vite as a dev dependency
- Configure Vite to serve the existing `index.html`
- Validate that all existing functionality works through Vite's dev server
- Confirm that `lib/scoring.js` can be imported directly by the frontend (shared module test)
- Configure Vite's production build output to be served by Express

**Why Vite:**
- Near-instant dev server (native ES modules, no bundling during development)
- Minimal config compared to Webpack
- Official recommended tooling for Vue 3
- Makes shared ES modules between frontend and backend trivial
- See full discussion in #18

**Acceptance Criteria:**
- [ ] `npm run dev` starts Vite dev server serving the existing app
- [ ] `npm run build` produces a production bundle
- [ ] Express serves the built output correctly
- [ ] `lib/scoring.js` is importable from the frontend
- [ ] No behavior change from the user's perspective

Part of the expansion plan: #18

@claude can you fix this from the [CI / build-and-test (pull_request)](https://github.com/dantonel/bridge-scorer/actions/runs/23578944498/job/68657576089?pr=29) step?

4s
Run npm ci
npm warn EBADENGINE Unsupported engine {
npm warn EBADENGINE   package: 'bridge-scorer@1.0.0',
npm warn EBADENGINE   required: { node: '22.x' },
npm warn EBADENGINE   current: { node: 'v24.14.0', npm: '11.9.0' }
npm warn EBADENGINE }
npm error code EUSAGE
npm error
npm error `npm ci` can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.
npm error
npm error Missing: vite@6.4.1 from lock file
npm error Missing: esbuild@0.25.12 from lock file
npm error Missing: fdir@6.5.0 from lock file
npm error Missing: fsevents@2.3.3 from lock file
npm error Missing: picomatch@4.0.4 from lock file
npm error Missing: postcss@8.5.8 from lock file
npm error Missing: rollup@4.60.0 from lock file
npm error Missing: tinyglobby@0.2.15 from lock file
npm error Missing: @esbuild/aix-ppc64@0.25.12 from lock file
npm error Missing: @esbuild/android-arm@0.25.12 from lock file
npm error Missing: @esbuild/android-arm64@0.25.12 from lock file
npm error Missing: @esbuild/android-x64@0.25.12 from lock file
npm error Missing: @esbuild/darwin-arm64@0.25.12 from lock file
npm error Missing: @esbuild/darwin-x64@0.25.12 from lock file
npm error Missing: @esbuild/freebsd-arm64@0.25.12 from lock file
npm error Missing: @esbuild/freebsd-x64@0.25.12 from lock file
npm error Missing: @esbuild/linux-arm@0.25.12 from lock file
npm error Missing: @esbuild/linux-arm64@0.25.12 from lock file
npm error Missing: @esbuild/linux-ia32@0.25.12 from lock file
npm error Missing: @esbuild/linux-loong64@0.25.12 from lock file
npm error Missing: @esbuild/linux-mips64el@0.25.12 from lock file
npm error Missing: @esbuild/linux-ppc64@0.25.12 from lock file
npm error Missing: @esbuild/linux-riscv64@0.25.12 from lock file
npm error Missing: @esbuild/linux-s390x@0.25.12 from lock file
npm error Missing: @esbuild/linux-x64@0.25.12 from lock file
npm error Missing: @esbuild/netbsd-arm64@0.25.12 from lock file
npm error Missing: @esbuild/netbsd-x64@0.25.12 from lock file
npm error Missing: @esbuild/openbsd-arm64@0.25.12 from lock file
npm error Missing: @esbuild/openbsd-x64@0.25.12 from lock file
npm error Missing: @esbuild/openharmony-arm64@0.25.12 from lock file
npm error Missing: @esbuild/sunos-x64@0.25.12 from lock file
npm error Missing: @esbuild/win32-arm64@0.25.12 from lock file
npm error Missing: @esbuild/win32-ia32@0.25.12 from lock file
npm error Missing: @esbuild/win32-x64@0.25.12 from lock file
npm error Missing: nanoid@3.3.11 from lock file
npm error Missing: picocolors@1.1.1 from lock file
npm error Missing: source-map-js@1.2.1 from lock file
npm error Missing: @rollup/rollup-android-arm-eabi@4.60.0 from lock file
npm error Missing: @rollup/rollup-android-arm64@4.60.0 from lock file
npm error Missing: @rollup/rollup-darwin-arm64@4.60.0 from lock file
npm error Missing: @rollup/rollup-darwin-x64@4.60.0 from lock file
npm error Missing: @rollup/rollup-freebsd-arm64@4.60.0 from lock file
npm error Missing: @rollup/rollup-freebsd-x64@4.60.0 from lock file
npm error Missing: @rollup/rollup-linux-arm-gnueabihf@4.60.0 from lock file
npm error Missing: @rollup/rollup-linux-arm-musleabihf@4.60.0 from lock file
npm error Missing: @rollup/rollup-linux-arm64-gnu@4.60.0 from lock file
npm error Missing: @rollup/rollup-linux-arm64-musl@4.60.0 from lock file
npm error Missing: @rollup/rollup-linux-loong64-gnu@4.60.0 from lock file
npm error Missing: @rollup/rollup-linux-loong64-musl@4.60.0 from lock file
npm error Missing: @rollup/rollup-linux-ppc64-gnu@4.60.0 from lock file
npm error Missing: @rollup/rollup-linux-ppc64-musl@4.60.0 from lock file
npm error Missing: @rollup/rollup-linux-riscv64-gnu@4.60.0 from lock file
npm error Missing: @rollup/rollup-linux-riscv64-musl@4.60.0 from lock file
npm error Missing: @rollup/rollup-linux-s390x-gnu@4.60.0 from lock file
npm error Missing: @rollup/rollup-linux-x64-gnu@4.60.0 from lock file
npm error Missing: @rollup/rollup-linux-x64-musl@4.60.0 from lock file
npm error Missing: @rollup/rollup-openbsd-x64@4.60.0 from lock file
npm error Missing: @rollup/rollup-openharmony-arm64@4.60.0 from lock file
npm error Missing: @rollup/rollup-win32-arm64-msvc@4.60.0 from lock file
npm error Missing: @rollup/rollup-win32-ia32-msvc@4.60.0 from lock file
npm error Missing: @rollup/rollup-win32-x64-gnu@4.60.0 from lock file
npm error Missing: @rollup/rollup-win32-x64-msvc@4.60.0 from lock file
npm error Missing: @types/estree@1.0.8 from lock file
npm error
npm error Clean install a project
npm error
npm error Usage:
npm error npm ci
npm error
npm error Options:
npm error [--install-strategy <hoisted|nested|shallow|linked>] [--legacy-bundling]
npm error [--global-style] [--omit <dev|optional|peer> [--omit <dev|optional|peer> ...]]
npm error [--include <prod|dev|optional|peer> [--include <prod|dev|optional|peer> ...]]
npm error [--strict-peer-deps] [--foreground-scripts] [--ignore-scripts]
npm error [--allow-git <all|none|root>] [--no-audit] [--no-bin-links] [--no-fund]
npm error [--dry-run]
npm error [-w|--workspace <workspace-name> [-w|--workspace <workspace-name> ...]]
npm error [--workspaces] [--include-workspace-root] [--install-links]
npm error
npm error aliases: clean-install, ic, install-clean, isntall-clean
npm error
npm error Run "npm help ci" for more info
npm error A complete log of this run can be found in: /home/runner/.npm/_logs/2026-03-26T05_24_21_293Z-debug-0.log
Error: Process completed with exit code 1.

--- a/.gitignore
+++ b/.gitignore
@@ -1,1 +1,2 @@
 node_modules/
+dist/

--- a/CLAUDE.md
+++ b/CLAUDE.md
@@ -56,6 +56,10 @@ Data is persisted in Redis.
 - Run `npm test` to execute the test suite
 - Do not break existing API contracts — the frontend depends on exact response shapes
 
+## Dependency Management
+- After adding or changing npm dependencies in `package.json`, always run `npm install` to update `package-lock.json`
+- Commit both `package.json` and `package-lock.json` together — CI uses `npm ci` which requires them to be in sync
+
 ## Things to Be Careful About
 - Redis connection is lazy and shared — don't create multiple clients
 - Game state uses deepMerge for PATCH updates — understand this before modifying update logic

--- a/app.js
+++ b/app.js
@@ -1,21 +1,26 @@
 import express from 'express';
+import { existsSync } from 'node:fs';
 import { handler } from './server.js';
 
 const app = express();
 const PORT = process.env.PORT || 3000;
 
+// Use Vite build output if available, otherwise fall back to source root
+const staticRoot = existsSync('./dist/index.html') ? './dist' : '.';
+console.log(`Serving static files from: ${staticRoot}`);
+
 // Parse JSON request bodies
 app.use(express.json());
 
-// Serve static files (your index.html etc)
-app.use(express.static('.'));
+// Serve static files (Vite dist/ if built, otherwise source root)
+app.use(express.static(staticRoot));
 
 // Route all /api requests to your existing handler
 app.all('/api/*', handler);
 
 // Fallback: serve index.html for everything else
 app.get('*', (req, res) => {
-  res.sendFile('index.html', { root: '.' });
+  res.sendFile('index.html', { root: staticRoot });
 });
 
 app.listen(PORT, () => {

--- a/package.json
+++ b/package.json
@@ -6,11 +6,17 @@
   "main": "app.js",
   "scripts": {
     "start": "node app.js",
-    "start:vercel": "node server.js"
+    "start:vercel": "node server.js",
+    "dev": "vite",
+    "build": "vite build",
+    "preview": "vite preview"
   },
   "dependencies": {
     "express": "^4.18.0",
     "redis": "^4.6.0"
+  },
+  "devDependencies": {
+    "vite": "^6.0.0"
   },
   "engines": {
     "node": "22.x"

--- a/vite.config.js
+++ b/vite.config.js
@@ -1,0 +1,13 @@
+import { defineConfig } from 'vite';
+
+export default defineConfig({
+  root: '.',
+  build: {
+    outDir: 'dist',
+  },
+  server: {
+    proxy: {
+      '/api': 'http://localhost:3000',
+    },
+  },
+});

